### DISCUSSION TITLE
Gemini batch input schema

### DISCUSSION THREAD

**Discussion by rpatidar** [GoogleCloudPlatform/generative-ai#1585] [Q&A]
I am looking for the gemini batch input api schema details. 

there are input example given [here](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/batch-prediction-gemini) but i assume this  

`{"request":{"contents": [{"role": "user", "parts": [{"text": "What is the relation between the following video and image samples?"}, {"fileData": {"fileUri": "gs://cloud-samples-data/generative-ai/video/animals.mp4", "mimeType": "video/mp4"}}, {"fileData": {"fileUri": "gs://cloud-samples-data/generative-ai/image/cricket.jpeg", "mimeType": "image/jpeg"}}]}]}}
`

but the how do i specifiy more attributes such as request id, or model parameters etc are not there. 

i don't see any details in  the examples uploaded as well. 

Relevant Community questions
https://www.googlecloudcommunity.com/gc/AI-ML/How-do-you-add-quot-parameters-quot-to-a-batch-prediction-job-s/m-p/623569 
https://www.googlecloudcommunity.com/gc/AI-ML/How-to-specify-the-Reference-Id-for-each-prompts-requests-in/m-p/851737

**Comment by holtskinner** [comment]
You can see some example of the input in the Notebook here https://github.com/GoogleCloudPlatform/generative-ai/blob/main/gemini/batch-prediction/intro_batch_prediction.ipynb

Each line needs to follow the request format listed here:

https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference#request
